File: Assets/Scripts/Model/Bullet.cs
using UnityEngine;
using System.Collections;

public class Bullet : MonoBehaviour 
{
    private float mAngle = 0;
    private Vector3 mDirection = Vector3.right;
    public float Angle
    {
        get { return mAngle; }
        set 
        {
            mAngle = value;
            transform.eulerAngles = new Vector3(0, 0, mAngle);
            mDirection = Quaternion.Euler(new Vector3(0, 0, mAngle)) * Vector3.right;
        }
    }

    private float mSpeed = 700;
    public float Speed
    {
        get { return mSpeed; }
        set { mSpeed = value; }
    }

	// Use this for initialization
	void Start () 
    {
	
	}
	
	// Update is called once per frame
	void Update ()
    {
        transform.localPosition += Time.deltaTime * mSpeed * mDirection;
        if (transform.localPosition.x < -640 || transform.localPosition.x > 640 || transform.localPosition.y < -360 || transform.localPosition.y > 360)
        {
            Destroy(gameObject);
            return;
        }
        Vector3 worldpos1 = UICamera.currentCamera.transform.TransformPoint(transform.localPosition);
        Vector3 pos = UICamera.currentCamera.WorldToScreenPoint(worldpos1);
        Ray ray = Camera.main.ScreenPointToRay(pos);
        if (Physics.Raycast(ray, 10000))
        {
            Bomb bomb = (Instantiate(Resources.Load("BombPrefabs/Bomb_01")) as GameObject).GetComponent<Bomb>();
            bomb.transform.parent = UIManager.GetInstance().WeaponRoot;
            bomb.transform.localPosition = transform.localPosition;
            bomb.transform.localScale = Vector3.one;
            bomb.BombAction();
            Destroy(gameObject);
        }
	}
}

File: Assets/Framework/Lua/Logic/Bullet.lua
Bullet = class("Bullet",require("Logic.NodeBase"))

function Bullet:ctor(obj)
    Bullet.super.ctor(self,obj)
    self.active_ = false
    self.dir_ = Vector3.right
    self.speed_ = 700
end

function Bullet:setSpeed(speed)
    self.speed_ = speed
end

function Bullet:OnUpdate(dt)
    self.transform_.localPosition = self.transform_.localPosition + self.dir_ * dt * self.speed_;
    if self.transform_.localPosition.x < -600 or self.transform_.localPosition.x > 600 or self.transform_.localPosition.y < -360 or self.transform_.localPosition.y > 300 then
        MainView:recycleBullet(self)
        return;
    end

    local worldpos1 = UICamera.currentCamera.transform:TransformPoint(self.transform_.localPosition)
    local pos = UICamera.currentCamera:WorldToScreenPoint(worldpos1)
    local ray = Camera.main:ScreenPointToRay(pos)
    local layerMask = 2 ^ LayerMask.NameToLayer("Fish")
    local flag,hitInfo = Physics.Raycast(ray, nil, 10000, layerMask)
    if flag then
        MainView:showBomb(self.transform_.localPosition)
        MainView:recycleBullet(self)
    end
end